templates/bonsai/contracts/tests/contract_tests/utils.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Utilities for writing tests for Bonsai applications. Supports testing of applications that use
+//! the Bonsai proxy to make requests for processing by a RISC Zero guest by providing utilities to
+//! run a mock of Bonsai.
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::future::Future;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use ethers::core::k256::ecdsa::SigningKey;
+use ethers::prelude::*;
+use ethers::utils::{Ganache, GanacheInstance};
+use risc0_zkvm::{Prover, ProverOpts};
+use tokio::sync::oneshot;
+use tokio::task::JoinHandle;
+
+abigen!(
+    MockBonsaiProxy,
+    "artifacts/MockBonsaiProxy.sol/MockBonsaiProxy.json"
+);
+
+pub type Client = Arc<SignerMiddleware<Provider<Ws>, Wallet<SigningKey>>>;
+
+pub async fn get_ganache_client() -> Result<(GanacheInstance, Client), Box<dyn Error>> {
+    // Launch Ganache instance
+    let ganache = Ganache::new().spawn();
+
+    // Instantiate wallet
+    let wallet: LocalWallet = ganache.keys()[0].clone().into();
+
+    // Connect to Ganache
+    let provider = Provider::<Ws>::connect(ganache.ws_endpoint()).await?;
+
+    // Instantiate client as wallet on network
+    let client = Arc::new(SignerMiddleware::new(
+        provider,
+        wallet.with_chain_id(1337u64),
+    ));
+
+    Ok((ganache, client))
+}
+
+pub struct BonsaiMock {
+    pub contract_address: Address,
+    pub task_handle: JoinHandle<()>,
+}
+
+impl BonsaiMock {
+    pub async fn spawn(
+        client: Client,
+        registry: HashMap<[u8; 32], &'static [u8]>,
```

Comment:
Might be good to use a type or at least a way to remove the magic length number.

```suggestion
        registry: HashMap<Digest, &'static [u8]>,
```

or

```suggestion
        registry: HashMap<ImageId, &'static [u8]>,
```

I'm also not sure it's a good idea to use the `'static` lifetime.